### PR TITLE
Modifying - IDENTITY-3323

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
Saving the app bean in the session with a UUID instead of using service provider name.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java [hunk 1]
```diff
@@ -19,14 +19,61 @@
...
+     * Remove related application bean from the session.
+     *
+     * @param session Http Session.
+     * @param spName  Service provider name.
+     */
+    public static void removeApplicationBeanFromSession(HttpSession session, String spName) {
+
+        Map<String, UUID> spUniqueIdMap = (HashMap<String, UUID>) session.getAttribute("spUniqueIdMap");
+
+        if (spUniqueIdMap.get(spName) == null) {
```
Comment (johannnallathamby) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: NullPointerException if Map is not there in session. Follow the same pattern in getApplicationBeanFromSession()

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java [hunk 2]
```diff
@@ -19,14 +19,65 @@
...
+     * @param session HTTP Session.
+     * @param spName  Service provider name.
+     * @return ApplicationBean
+     */
+    public static ApplicationBean getApplicationBeanFromSession(HttpSession session, String spName) {
+
+        Map<String, UUID> spUniqueIdMap;
+
+        if (session.getAttribute("spUniqueIdMap") == null) {
+
```
Comment (hasinthaindrajee) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: Remove unnecessary new lines

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/remove-service-provider.jsp [hunk 3]
```diff
@@ -56,11 +57,13 @@
 				String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
 				String backendServerURL = CarbonUIUtil.getServerURL(config.getServletContext(), session);
 				ConfigurationContext configContext =
-				                                     (ConfigurationContext) config.getServletContext()
-				                                                                  .getAttribute(CarbonConstants.CONFIGURATION_CONTEXT);
+						(ConfigurationContext) config.getServletContext()
```
Comment (hasinthaindrajee) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/remove-service-provider.jsp]: Format code.

Comment (JKAUSHALYA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/remove-service-provider.jsp]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java (hunk 2 above)
Comment (JKAUSHALYA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java [hunk 4]
```diff
@@ -19,14 +19,65 @@
...
             ApplicationBean applicationBean = new ApplicationBean();
-            session.setAttribute(spName, applicationBean);
+            UUID uuid = UUID.randomUUID();
+
+            spUniqueIdMap.put(spName, uuid);
+            session.setAttribute(uuid.toString(), applicationBean);
         }
-        return (ApplicationBean)session.getAttribute(spName);
+
+        return (ApplicationBean) session.getAttribute(spUniqueIdMap.get(spName).toString());
```
Comment (mohanvive) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: Most of the places you have unnecessary spaces..

Comment (JKAUSHALYA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java [hunk 5]
```diff
@@ -19,14 +19,65 @@
...
+     *
+     * @param session HTTP Session.
+     * @param spName  Service provider name.
+     * @return ApplicationBean
+     */
+    public static ApplicationBean getApplicationBeanFromSession(HttpSession session, String spName) {
+
+        Map<String, UUID> spUniqueIdMap;
+
+        if (session.getAttribute("spUniqueIdMap") == null) {
```
Comment (sumuditha-viraj) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: We can see "spUniqueIdMap" string used in few places in this class. Can we define a String constant for "spUniqueIdMap"

Comment (JKAUSHALYA) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIUtil.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

**Pull request merged**